addon/helpers/route-idle.js

```diff
@@ -1,17 +1,32 @@
 import Helper from '@ember/component/helper';
+import { inject as service } from '@ember/service';
 import { tracked } from '@glimmer/tracking';
 import { whenRouteIdle } from '../scheduler';
 
 export default class RouteIdle extends Helper {
+  @service router;
+
   @tracked isIdle = false;
 
+  constructor() {
+    super(...arguments);
+
+    this.router.on('routeWillChange', () => {
+      if (!this.isDestroying) {
+        this.isIdle = false;
+      }
+    });
+  }
+
   compute() {
     if (this.isIdle) {
       return true;
     }
 
     whenRouteIdle().then(() => {
-      this.isIdle = true;
+      if (!this.isDestroying) {
+        this.isIdle = true;
+      }
     });
```

Comment:
Instead of manually subscribing, we should be able to leverage auto-tracking for this. See https://github.com/rwjblue/ember-router-helpers/pull/263 for how I've done it in the past.

Reply:
Thanks for calling that out! I'm still getting the hang of auto-tracking as a paradigm-- refactored to make this component much more elegant.